src/ae.c

```diff
@@ -63,6 +62,17 @@
 #endif
 #endif
 
+#define AE_LOCK(eventLoop)                                                                                             \
+    int __ae_locked = 0;                                                                                               \
+    if ((eventLoop)->flags & AE_PROTECT_POLL) {                                                                        \
+        pthread_mutex_lock(&(eventLoop)->poll_mutex);                                                                  \
```

Comment:
Small suggestion (I know we did not do it in other places): in this implementation you do not plan for recursive mutex calls (which is correct). however since we are using an unsafe form of lock/unlock via goto jumps, maybe it would be good to protect this guarantee and compile the Valkey so the mutex will be created with PTHREAD_MUTEX_ERRORCHECK  attribute. we can then assert when the mutext lock/unlock calls fail.
What do you think?

Reply:
Thanks, changed to use `PTHREAD_MUTEX_ERRORCHECK`.